File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.c
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize){
    int right = numbersSize - 1;
    int left = 0;
    int *k = (int*)malloc(sizeof(int) * 2);
    *returnSize = 2;
    
    k[0] = -1;
    while(k[0] == -1)
    {
        if(numbers[left] + numbers[right] > target)
            right--;
        else if(numbers[left] + numbers[right] < target)
            left++;
        else
        {
            k[0] = left + 1;
            k[1] = right + 1;
        }
    }
    return k; 
}

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] ans = new int[2];
        int l = 0;
        int r = numbers.length-1;
        
        while(l <= r){
            if(numbers[l] + numbers[r] == target){
                ans[0] = l+1;
                ans[1] = r+1;
                l = r+1;
            }
            else if(numbers[l] + numbers[r] > target)
                r--;
            else
                l++;
        }
        
        return ans;
    }
}